src/execution_plan/optimizations/traverse_order_utils.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2018-2020 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#include "RG.h"
+#include "../../util/arr.h"
+#include "../../util/strcmp.h"
+#include "traverse_order_utils.h"
+
+//------------------------------------------------------------------------------
+// Scoring functions
+//------------------------------------------------------------------------------
+
+// score expression based on its source and destination labels
+// score = src label count + dest label count
```

Comment:
Referring to label count is misleading, as `QGNode_LabelCount` can only evaluate to 0 or 1. Either fix that function or rewrite names and comments to reflect `IsLabeled` rather than `LabelCount`.

Reply:
the intention here is to be compatible with the upcoming support for multi-label